diff --git a/debian/rules b/debian/rules
index 3d12fb5..f9962ac 100755
--- a/debian/rules
+++ b/debian/rules
@@ -38,6 +38,13 @@ export DO_LPF=1
 CONFFLAGS+=--enable-use-sockets
 endif
 
+ifeq (,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+  PARALLEL :=
+else
+  PARALLEL := \
+       -j$(patsubst parallel=%,%,$(filter parallel=%,$(DEB_BUILD_OPTIONS)))
+endif
+
 %:
 	dh $@ --with autoreconf
 
@@ -62,19 +69,19 @@ override_dh_auto_build:
 	# ldap-enabled build
 	test -f Makefile && $(MAKE) distclean || true
 	./configure --with-ldap --with-ldapcrypto CFLAGS="$(CFLAGS) -DNSUPDATE" $(CONFFLAGS) LIBS="$(LIBS) -latomic"
-	$(MAKE)
+	$(MAKE) $(PARALLEL)
 	mv server/dhcpd dhcpd
 	# ddns-disabled build
 	test -f Makefile && $(MAKE) distclean || true
 	./configure CFLAGS="$(CFLAGS)" $(CONFFLAGS) LIBS="$(LIBS) -latomic"
-	$(MAKE)
+	$(MAKE) $(PARALLEL)
 	mv client/dhclient dhclient
 	# ldap-disabled build
 	test -f Makefile && $(MAKE) distclean || true
 	./configure CFLAGS="$(CFLAGS) -DNSUPDATE" $(CONFFLAGS) LIBS="$(LIBS) -latomic"
-	$(MAKE)
+	$(MAKE) $(PARALLEL)
 	# keama build
-	$(MAKE) -C keama
+	$(MAKE) $(PARALLEL) -C keama
 
 override_dh_auto_test:
 	/bin/sh keama/tests/runall.sh 2> /tmp/keama-test-errors
-- 
2.25.1

